core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -284,7 +284,7 @@ public int register(String subject,
 
       // see if the schema to be registered already exists
       AvroSchema avroSchema = canonicalizeSchema(schema);
-      MD5 md5 = MD5.ofString(schema.getSchema());
+      MD5 md5 = MD5.ofString(schema.getSchema() + subject);
```

Comment:
This would definitely be a backwards incompatible change.

I think I'm confused about the solution here anyway though. Isn't the problem identified in the issue discussion that the client-side caching isn't correct? So I don't think we should need to change anything in the core schema registry service code. Wouldn't the solution just be to fix the key used when caching the schema to include the subject?